### PR TITLE
[Configurable Autosubmit] Update revert validation to use configurable autosubmit configuration - Part 5

### PR DESCRIPTION
**Pull request by ricardoamador**
Update the revert validation to use the new repository configuration values when validating a revert request.

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: auto_submit/lib/validations/revert.dart [hunk 1]
```diff
@@ -74,14 +73,14 @@ class Revert extends Validation {
     final github.RepositorySlug repositorySlug = _getSlugFromLink(revertLink);
     githubService = await config.createGithubService(repositorySlug);
 
-    final bool requiredChecksCompleted = await waitForRequiredChecks(
-      githubService: githubService,
-      slug: repositorySlug,
-      sha: sha!,
-      checkNames: requiredCheckRunsMapping[repositorySlug.name]!,
+    ///TODO(ricardoamador) this should be moved out to the main validations class as a separate check.
```
Comment (keyonghan) [inline auto_submit/lib/validations/revert.dart:76]:
```suggestion
    // TODO(ricardoamador) this should be moved out to the main validations class as a separate check.
```

**Review by keyonghan — APPROVED** [review]
LGTM with nits.

File: auto_submit/lib/validations/required_check_runs.dart [hunk 2]
```diff
@@ -5,15 +5,108 @@
...
+class RequiredCheckRuns extends Validation {
+  const RequiredCheckRuns({
+    required super.config,
+    RetryOptions? retryOptions,
+  }) : retryOptions = retryOptions ?? Config.requiredChecksRetryOptions;
+
+  final RetryOptions retryOptions;
+
+  Future<bool> waitForRequiredChecks({
+    // required GithubService githubService,
```
[the remark this answers is not in the document]
Comment (ricardoamador) [reply auto_submit/lib/validations/required_check_runs.dart:33]: Done.

File: auto_submit/lib/validations/revert.dart (hunk 1 above)
Comment (ricardoamador) [reply auto_submit/lib/validations/revert.dart:76]: Fixed.

**Comment by ricardoamador** [comment]
Yeah that was due to the github outage.

**Pull request merged**